srtcore/core.h

```diff
@@ -1851,5 +1123,40 @@ class CUDT
     void removeEPollID(const int eid);
 };
 
+// Socket option value casting functions
+template <typename T>
+inline T cast_optval(const void* optval)
+{
+    return *reinterpret_cast<const T*>(optval);
+}
+
+template <typename T>
+inline T cast_optval(const void* optval, int optlen)
+{
+    if (optlen > 0 && optlen != sizeof(T))
+        throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
+
+    return cast_optval<T>(optval);
+}
```

Comment:
Moved these functions to core.h, as they are used both in core.cpp and socket_group.cpp.